config/tests/config_tests.rs

```diff
@@ -29,6 +29,24 @@ use std::{fs::File, io::Write};
 use tempfile::tempdir;
 use test_utils::make_authority_with_port_getter;
 
+#[test]
+fn leader_election_rotates_through_all() {
+    // this committee has equi-sized stakes
+    let committee = test_utils::committee(None);
+    let mut leader_counts = HashMap::new();
+    let mut leader_counts_stepping_by_2 = HashMap::new();
+    for i in 0..100 {
+        let leader = committee.leader(i);
+        let leader_stepping_by_2 = committee.leader(i * 2);
+        *leader_counts.entry(leader).or_insert(0) += 1;
+        *leader_counts_stepping_by_2
+            .entry(leader_stepping_by_2)
+            .or_insert(0) += 1;
+    }
+    assert!(leader_counts.values().all(|v| *v > 1));
```

Comment:
Since we are weighting based on stake would it make sense to also check the diff between the number of times the leaders have been elected and ensure they don't diverge much? (since we use normal distribution).

Reply:
It would be an improvement, but I don't see as a main . 
We use a uniform distribution for the `test_utils::committee` function, which means we are doing Bernouilli sampling. 

If we want to be as general as we can, this test should generate a binomial distribution, something we can test [with a goodness-of-fit test adapted for the discrete case](http://www.stat.yale.edu/~jay/EmersonMaterials/DiscreteGOF.pdf). We can also test exact numbers (the test is deterministic), but this may break the second we get out of our current discipline of iterating through seeds in a particular way. A good trade-off would be to use a simple inequality (e.g. Chebysheff).

I'll open an issue & follow-up.

Reply:
Thanks for the detailed reply on this! Great to see that you did with solution (2). Definitely (3) is something we could apply here - and now that I can think off we could use in other places in the end to end tests .